my_compassion/controllers/my2_children.py

```diff
@@ -0,0 +1,76 @@
+##############################################################################
+#
+#    Copyright (C) 2025 Compassion CH (http://www.compassion.ch)
+#    @author: Noé Berdoz <nberdoz@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo.http import request
+from odoo import http
+
+
+class MyCompassionChildrenController(http.Controller):
+
+    @http.route('/my2/children/', type="http", auth="user", website=True)
+    def my2_render_children_page(self, **kwargs):
+        """
+        Renders the children page related to the logged-in user's sponsorships.
+        return: An HTTP response containing a rendered template with sponsorship data.
+        """
+        partner = request.env.user.partner_id
+
+        breadcrumbs = [
+            {'name': 'Children', 'url': '/my2/children/', 'active': True},
+        ]
+
+        return request.render(
+            'my_compassion.my2_children_page',
+            {
+                'sponsorship_ids': partner.sponsorship_ids,
+                'breadcrumbs': breadcrumbs,
+            }
+        )
+
+    @http.route('/my2/children/<int:child_id>', type="http", auth="user", website=True)
+    def my2_render_child_timeline_page(self, child_id, **kwargs):
```

Comment:
Directly fetching the child record will also use the security access rules logic to make sure the user has the rights to read information of the selected child.
```suggestion
    @http.route('/my2/children/<model('compassion.child'):child>', type="http", auth="user", website=True, sitemap=False)
    def my2_render_child_timeline_page(self, child, **kwargs):
```

Reply:
Thank you, I created a related task for an overall checking where this optimization is possible, task [T2348](https://erp.compassion.ch/web#id=2940&action=521&active_id=901&model=project.task&view_type=form&cids=1&menu_id=2029)